Connectwizard regression fix and related fixes and refinements

rserwin1 opened the pull request:
This commit corrects a recent change to connectwizard and makes some minor changes for probably-rare cases.  It removes a dependency on 'ethtool', allows control of ethernet connections when no network managers have been used, and may reduce the incidence of invalid IP addresses if MAC spoofing is in effect.  Details:
 - Connectwizard windows are centered on screen (as in FossaPup), instead of cascaded from upper left.
 - Connectwizard no longer sets current manager name for pgprs because pgprs now changes it only when connecting, allowing viewing of its setup without making it "current".
 - Connectwizard and network_default_connect accommodate connman alternate network manager.
 - The default ethernet interface (used when no network manager has been set up) can now be disconnected and reconnected with the network tray icon.
 - The default ethernet setup no longer uses ethtool to check for "plugged in", replaced by ip-show and a test for LOWER_UP.
 - The default ethernet setup waits 10 seconds longer for an IP address, which may reduce the incidence of failure (indicated by an address of 169.254...), possibly more likely if MAC address spoofing is used on a LAN.
 - All components have been checked with 'shellcheck' and have had its warnings resolved, making some of the code more efficient, simpler and less vulnerable to coding mistakes.

rizalmart committed:
z_update_system_cache.sh: Added udev rule reload

Reload udev rule if the packages contains udev rules

dimkr committed:
backport puppylinux-woof-CE/gtkdialog#126

rserwin1 committed:
Connectwizard regression fix and related fixes and refinements

dimkr committed:
auto-configure applications to run as spot when .desktop is added

dimkr committed:
Merge pull request #2846 from dimkr/feature/auto-spot

Auto-configure applications to run as spot when executable is created

dimkr committed:
fix breakage of sid64 after the transition to libwacom9

dimkr committed:
Merge pull request #2850 from dimkr/bugfix/sid-libwacom

Fix breakage of sid64 after the transition to libwacom9

dimkr committed:
pin the versions of yambar dependencies

dimkr committed:
Merge pull request #2851 from dimkr/bugfix/pin-yambar-deps

Pin the versions of yambar dependencies

dimkr commented on woof-code/rootfs-skeleton/etc/rc.d/rc.network_eth:
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
```
@rserwin1 Can you reopen this PR after removing all the whitespace/indentation changes? This code is very messy, and it's hard to review it when most changes are cosmetic.

dimkr committed:
add GTK+ 3 support to Buntoo

dimkr committed:
fix font inconsistency between the Buntoo menu and everything else

peabee committed:
All front_change to finish even if desktop is not ROX-Filer

see #2852

dimkr committed:
use black window outline in 412

dimkr committed:
Merge pull request #2858 from dimkr/bugfix/412-outline

Use black window outline in 412

dimkr committed:
Merge pull request #2854 from dimkr/feature/buntoo-gtk3

Add GTK+ 3 support to Buntoo

dimkr committed:
add the 605 global theme

dimkr committed:
Merge pull request #2853 from dimkr/feature/605-theme

Add the 605 global theme

dimkr committed:
fix breakage of bionic64 introduced in 2bf56f5

dimkr committed:
Merge pull request #2860 from dimkr/bugfix/bionic64-no-make

Fix breakage of bionic64 introduced in 2bf56f5

Commit:
Modify buntoo wallpaper

Commit:
Add Stardust* openbox themes

dimkr committed:
fix broken instructions in dpup release notes

dimkr committed:
Merge pull request #2865 from dimkr/bugfix/dpup-hinting-lies

Fix broken instructions in dpup release notes

dimkr committed:
add /usr merge compatible kernel packages

peabee committed:
remove redundant code

as per @dimkr review

peabee committed:
Merge pull request #2856 from puppylinux-woof-CE/peabee-patch-1

All front_change to finish even if desktop is not ROX-Filer

rserwin1 commented:
Instead of undoing some of my effort, I would prefer to submit another PR with only the indentation changes and, if helpful, the replacement of "`...`" with "$(...)".  You could merge that and then examine the remaining changes in the current PR.  OK?

dimkr commented:
That would make it easier to review the changes. But @01micko should decide whether or not to merge, I'm just an extra pair of eyes looking at the code.

dimkr committed:
change to remove_sublevel=no and let DKMS rebuild out-of-tree drivers

dimkr committed:
add bdrv to automated releases

dimkr committed:
Merge pull request #2869 from dimkr/feature/bdrv-in-release

Add bdrv to automated releases

dimkr committed:
if present, add the kernel sources SFS to ext4 images

dimkr committed:
Merge pull request #2868 from dimkr/feature/dkms-support

Change to remove_sublevel=no and let DKMS rebuild out-of-tree drivers

dimkr committed:
Merge pull request #2870 from dimkr/feature/kernel-sources-in-image

If present, add the kernel sources SFS to ext4 images

dimkr committed:
Merge pull request #2866 from dimkr/feature/kernel-kit-usrmerge

Add /usr merge compatible kernel packages

dimkr committed:
update dpup release notes

dimkr committed:
Merge pull request #2871 from dimkr/cleanup/dpup-release-notes

Update dpup release notes

dimkr committed:
fix breakage of ARM kernel builds

dimkr committed:
Merge pull request #2872 from dimkr/bugfix/arm-breakage

Fix breakage of ARM kernel builds

Commit:
Merge pull request #2862 from 01micko/stardust-obthemes

Add Stardust* openbox themes

Commit:
Merge pull request #2861 from 01micko/buntoo-wall

Modify buntoo wallpaper

dimkr committed:
add a tiny kernel sources SFS for building out-of-tree drivers

dimkr committed:
rename "605" to "Tahrpup" and give it a nicer wallpaper

dimkr committed:
Merge pull request #2874 from dimkr/feature/small-kernel-sources

Add a tiny kernel sources SFS for building out-of-tree drivers

dimkr committed:
run init scripts immediately on x86_64 and optimize rc.services

dimkr committed:
restore missing delay on x86

dimkr committed:
Merge pull request #2876 from dimkr/bugfix/late-connman-start

Run init scripts immediately on x86_64 and optimize rc.services

dimkr committed:
Merge pull request #2877 from dimkr/bugfix/distro-arch-var

Restore missing delay on x86

dimkr committed:
Merge pull request #2878 from dimkr/bugfix/backport-gtkdialog-126

Backport puppylinux-woof-CE/gtkdialog#126

dimkr committed:
check the firewall state before the tray icon appears

dimkr committed:
update Xwayland to 22.1.0 (labwc)

dimkr committed:
update Xwayland to 22.1.0 (Cage)

dimkr committed:
fix immediate exit of the terminal when starting the firewall

dimkr committed:
Merge pull request #2882 from dimkr/bugfix/firewall-exit-immediately

Fix immediate exit of the terminal when starting the firewall

dimkr committed:
Merge pull request #2881 from dimkr/feature/xwayland-22.1

Update Xwayland to 22.1.0

dimkr committed:
update the firewall tray icon when /etc/init.d/rc.firewall runs and poll infrequently

dimkr committed:
Merge pull request #2879 from dimkr/bugfix/firewallstatus-always-false

Check the firewall state before the tray icon appears

dimkr committed:
Merge pull request #2880 from dimkr/feature/firewallstatus-no-polling

Update the firewall tray icon when /etc/init.d/rc.firewall runs

rserwin1 commented:
I sent dimkr a difference listing versus a similarly indented version of the current versions of the changed files.  If necessary, I can send the file sets for him to compare with, say, kdiff3.
In the meantime, shall I add a commit to make a minor change to network_default_connect?  The listing revealed a mistake in the indenting, which could be ignored if a new commit would delay the merging.  The change may not matter if/when we move these files to a new rootfs-package, or can be made at that time (if the connman code is retained).
Please tell me if you need anything more from me to allow the merge.

dimkr commented:
Even after hiding whitespace (https://github.com/puppylinux-woof-CE/woof-CE/pull/2848/files?diff=unified&w=1), I find this PR hard to review. I don't understand some changes, like the DHCP release thing.

dimkr committed:
Merge pull request #2855 from dimkr/bugfix/buntoo-huge-menu

Fix font inconsistency between the Buntoo menu and everything else

peabee committed:
CONFIG_USB_UHCI_HCD=y

Both 5.10 & 5.15 configs - needed for wireless keyboards during boot savearea selection

peabee committed:
Update DOTconfig-5.15.4-x86_64

peabee committed:
Update DOTconfig-5.10.60-i686-pae

peabee committed:
Update DOTconfig-5.15.4-i686-pae

dimkr committed:
Merge pull request #2875 from dimkr/feature/tahrpup-wallpaper

Rename "605" to "Tahrpup" and give it a nicer wallpaper

dimkr committed:
Merge pull request #2885 from puppylinux-woof-CE/peabee-patch-1

CONFIG_USB_UHCI_HCD=y

dimkr committed:
add Ubuntu bdrv support

dimkr committed:
use relative paths in efilinux.cfg

dimkr committed:
add gdebi to bdrv

rserwin1 commented:
I assume you refer to the dhcpcd-related changes in rc.network_eth.
It is actually only one change, to increase the dhcpcd timeout value to wait for a DNS to respond with a valid IP address, but only if a timeout setting is not in /etc/dhcpcd.conf.

Those changes would be unnecessary if we simply add a line, "timeout 40", to dhcpcd.conf.  That would be the best over-all fix for the problem, which I will explain.

While testing my changes for SNS-3.3, I found that the MAC address spoofing implementation caused my router to get slower in responding to dhcpcd after rebooting the PC a few times.  My other PC on the same LAN/router also failed to connect due to the "169.254" responses.  Each reboot generates a new spoofed MAC address, possibly necessitating the router to check a longer MAC list.  If dhcpcd timed out it would return an invalid IP address, 169.254...  A day after the spoofing, the invalid IP addresses did not occur, probably because the leases had expired.

After implementing retrying dhcpcd once, which usually then succeeded, it struck me that the better solution is to increase the timeout value.  But I wanted to confine that to SNS, even though the slowdown impacts any device on the same LAN no matter which network manager is in control.

I now believe that this PR should include the "timeout 40" addition to dhcpcd.conf.  BTW, the "40" is arbitrary, one third longer than the default of 30 seconds.  That way the "--timeout" logic can be removed from rc.network_eth (and SNS 3.3).  I request permission to make those (non-SNS) changes under this PR.

Edit:
I see that changing dhcpcd.conf is more complicated than I anticpated.  It is in the dhcpcd package.  Since I think we should be changing the content of that package, is there a standard way to update a configuration file during a build or at installation or boot-up time?

dimkr committed:
allow spot to install packages using gdebi

dimkr committed:
add basic IPv6 support to the firewall

dimkr committed:
add ::1 to /etc/hosts

dimkr committed:
keep ::1 in /etc/hosts after quicksetup

dimkr committed:
move ip6tables outside of devx

dimkr committed:
make pupswap work under PUPMODE 6

mavrothal committed:
Merge pull request #2893 from dimkr/bugfix/pupswap-wrong-path

Make pupswap work under PUPMODE 6

dimkr committed:
Merge pull request #2892 from dimkr/feature/basic-ipv6

Basic IPv6 support

dimkr committed:
Merge pull request #2889 from dimkr/bugfix/ext4-harcoded-drive

Use relative paths in efilinux.cfg

dimkr committed:
do not build bdrv when using pet packages

dimkr committed:
backport puppylinux-woof-CE/gtkdialog#133

dimkr committed:
Merge pull request #2895 from dimkr/bugfix/backport-gtkdialog-133

Backport puppylinux-woof-CE/gtkdialog#133

peabee committed:
frugalpup-37 update

dimkr committed:
Merge pull request #2886 from dimkr/feature/ubuntu-bdrv

Add Ubuntu bdrv support

peabee committed:
Add support for extra processing defined in _00build.conf

e.g. for devx manipulation

dimkr committed:
Merge pull request #2887 from dimkr/feature/gdebi-support

Add gdebi to bdrv

dimkr committed:
add command-not-found support to bdrv

peabee committed:
Update 3builddistro

rizalmart committed:
Update z_update_system_cache.sh

Commit:
Merge pull request #2793 from rizalmart/patch-406

z_update_system_cache.sh: Added udev rule reload

Commit:
Merge pull request #2896 from puppylinux-woof-CE/peabee-patch-1

Add support for extra processing defined in _00build.conf

dimkr committed:
Merge pull request #2884 from dimkr/feature/command-not-found

Add command-not-found support to bdrv

dimkr committed:
Merge pull request #2898 from dimkr/bugfix/cage-monitor-off

Disable set_wl_monsleep when using X

dimkr committed:
disable set_wl_monsleep when using X

dimkr committed:
Merge pull request #2900 from dimkr/bugfix/setup-spot-copy

Make setup-spot faster and safer

dimkr committed:
make setup-spot faster and safer

Commit:
labwc: update keyboard configuration re: #2903

This adds labwc support to /usr/sbin/xkbconfigmanager
Also, it fixes a bug where if a variant doesn't exist for a layout
the Xdialog gui crashes because of no input. Instead a message pops saying
that no variant is available for the layout. An example is au (Australia).

dimkr committed:
show only one menu entry for each .desktop file

Commit:
xorgwizard: exec wmons.sh on xrandrshell call for WAYLAND_DISPLAY

re #2903

dimkr committed:
show only one labwc menu entry for each .desktop file (fixes #2901)

Commit:
Make report-video more wayland compatible

re #2903

dimkr committed:
Merge pull request #2899 from dimkr/bugfix/many-gimp-vlc

Show only one menu entry for each .desktop file

dimkr committed:
Merge pull request #2902 from dimkr/bugfix/many-gimp-vlc-labwc

Show only one labwc menu entry for each .desktop file

dimkr committed:
Merge pull request #2907 from dimkr/bugfix/duplicate-labwc-menu-entries-2

Really remove duplicate labwc menu entries

Commit:
Merge pull request #2906 from 01micko/report-video-wayland

Make report-video more wayland compatible

dimkr committed:
Merge pull request #2908 from dimkr/bugfix/pkexec-as-root

Fix breakage of the Synaptic menu entry

Commit:
Merge pull request #2905 from 01micko/xorgwizard-wmon

xorgwizard: exec wmons.sh on xrandrshell call for WAYLAND_DISPLAY

Commit:
Merge pull request #2904 from 01micko/bugfix-keyboard

labwc: update keyboard configuration re: #2903

Commit:
box_splash: fix wayland geometry warning

re #2903

dimkr committed:
really remove duplicate labwc menu entries

dimkr committed:
fix breakage of the Synaptic menu entry

dimkr committed:
make it possible to run VS Code as spot (fixes #2909)

dimkr committed:
auto-configure VS Code to run as spot

Commit:
Merge pull request #2911 from 01micko/box_splash-wayland

box_splash: fix wayland geometry warning

dimkr committed:
add support for UEFI-bootable ISOs using cd-boot-images-amd64

dimkr committed:
fix breakage of Firefox

dimkr committed:
Merge pull request #2913 from dimkr/bugfix/firefox-titlebar

Restore the Firefox titlebar until joewing/jwm#540 is fixed

dimkr committed:
restore the Firefox titlebar until joewing/jwm#540 is fixed

dimkr committed:
Merge pull request #2912 from dimkr/feature/ubuntu-efi

Add support for UEFI-bootable ISOs using cd-boot-images-amd64

dimkr committed:
Merge pull request #2910 from dimkr/bugfix/vscode-as-spot

Make the VS Code .deb package work out-of-the-box

dimkr committed:
Merge pull request #2916 from dimkr/cleanup/dpup-llvm

Drop libllvm11 from sid64

dimkr committed:
Merge pull request #2915 from dimkr/hack/allow-debian-repos-in-upup

Add the -updates repo only for $DISTRO_BINARY_COMPAT:$DISTRO_COMPAT_VERSION

dimkr committed:
Merge pull request #2919 from dimkr/feature/ci-release-set-name

Allow the user to set the release DISTRO_NAME

dimkr committed:
add the -updates repo only for $DISTRO_BINARY_COMPAT:$DISTRO_COMPAT_VERSION

dimkr committed:
drop libllvm11 from sid64

dimkr committed:
ignore /etc/profile.d/*.csh

dimkr committed:
clear FIXUPHACK_FILE before processing a package

dimkr committed:
allow the user to set the release DISTRO_NAME

dimkr committed:
shrink cd-boot-images-amd64 and move it to devx

dimkr committed:
Merge pull request #2920 from dimkr/cleanup/cd-boot-images-devx

Shrink cd-boot-images-amd64 and move it to devx

dimkr committed:
Merge pull request #2918 from dimkr/bugfix/same-template-if-null

Clear FIXUPHACK_FILE before processing a package

dimkr committed:
Merge pull request #2917 from dimkr/bugfix/etc-profiled-csh

Ignore /etc/profile.d/*.csh

Commit:
petbuilds/geany/defaults.patch: use defaultterminal

`urxvt` may not be available, especially on wayland

Commit:
Merge pull request #2921 from 01micko/geany-patch-fix

petbuilds/geany/defaults.patch: use defaultterminal

dimkr committed:
move large Perl libraries to nlsx

Phoenix-Starlight committed:
Update Evil_blue_glow-jwmrc

Fix XML syntax.

dimkr committed:
drop the second copy of libgl1-mesa-dri in dpup and unify with libllvm

dimkr committed:
when building an ISO image, unify bdrv and rootfs-complete

Commit:
Merge pull request #2925 from Phoenix-Starlight/testing

Fix Evil_blue_glow-jwmrc

dimkr committed:
Merge pull request #2923 from dimkr/cleanup/perl-large-encode

Move large Perl encoding libraries to nlsx

dimkr committed:
Merge pull request #2926 from dimkr/cleanup/dpup-mesa-template

Drop the second copy of libgl1-mesa-dri in dpup and unify with libllvm

dimkr committed:
disallow installation of snapd

dimkr committed:
shrink bdrv

dimkr committed:
use the KDE cursor in the Tahrpup theme

dimkr committed:
Merge pull request #2932 from dimkr/feature/fossa-cursor

Use the KDE cursor in the Tahrpup theme

dimkr committed:
Merge pull request #2929 from dimkr/bugfix/broken-snapd

Disallow installation of snapd

dimkr committed:
Merge pull request #2930 from dimkr/cleanup/shrink-bdrv

Shrink bdrv

dimkr committed:
add a "rox" placeholder, if using pcmanfm

dimkr committed:
if not using ROX-Filer, add pmount to the Places menu

dimkr committed:
Merge pull request #2935 from dimkr/feature/places-menu-pmount

If not using ROX-Filer, replaces the Drives menu with pmount

rizalmart committed:
installpkg.sh

* Move /root/.packages to /var/packages
* Some code optimizations

rizalmart committed:
fixed syntax error

rizalmart committed:
removepreview.sh: Some improvements

* move /root/.packages to /var/packages
* some tweaks

rizalmart committed:
z_update_system_cache.sh: Some bugfixes

rizalmart committed:
hacks-postinstall2.sh: Some fixes

dimkr committed:
Merge pull request #2934 from dimkr/feature/rox-wrapper

Add a "rox" placeholder, if not using ROX-Filer

dimkr committed:
Merge pull request #2927 from dimkr/feature/bdrv-in-iso

When building an ISO image, unify bdrv and rootfs-complete

dimkr committed:
fix collision between pcmanfm and rootfs-hacks.sh

dimkr committed:
Merge pull request #2941 from dimkr/bugfix/rox-rootfs-hacks

Fix collision between pcmanfm and rootfs-hacks.sh

dimkr committed:
add pup-volume-monitor petbuild

dimkr committed:
switch to pcmanfm in bullseye

dimkr committed:
switch to prebuilt initrd.gz binaries from GitHub

dimkr committed:
move petbuild gio modules to the right location

dimkr committed:
show the pcmanfm sidebar by default

dimkr committed:
Revert "if not using ROX-Filer, add pmount to the Places menu"

This reverts commit a4b57f1520cc9974798ab6f693292dac6fbc5c54.

dimkr committed:
make the pcmanfm menu entry more consistent with others

dimkr committed:
change the default ConnMan settings

dimkr committed:
Merge pull request #2945 from dimkr/feature/connman-defaults

Change the default ConnMan settings

rizalmart committed:
removepreview.sh: Added package rollback

If the user package was removed but the package was part of puppy main SFS. Just remove the whiteout files to restore the removed builtin files

dimkr committed:
remove development packages from bdrv

dimkr committed:
remove devx from ext4 images, if bdrv is included

dimkr committed:
if using ConnMan, prevent psync from synchronizing the clock on boot

dimkr committed:
Merge pull request #2947 from dimkr/cleanup/psync-connmand-collision

If using ConnMan, prevent psync from synchronizing the clock on boot

dimkr committed:
Merge pull request #2944 from dimkr/cleanup/pcmanfm-menu-entry

Make the pcmanfm menu entry more consistent with others

dimkr committed:
Merge pull request #2943 from dimkr/feature/pup-volume-monitor-petbuild

Add pup-volume-monitor petbuild and restore the pcmanfm sidebar

dimkr committed:
remove more packages from bdrv

dimkr committed:
remove even more packages from bdrv

dimkr committed:
Merge pull request #2931 from dimkr/cleanup/bdrv-without-dev

Remove development packages from bdrv

dimkr committed:
if only Xwayland is present, start Xwayland and hide the X.Org button

dimkr committed:
when building an Ubuntu bdrv, use the default apt settings

dimkr committed:
Merge pull request #2949 from dimkr/feature/bdrv-ubuntu-defaults

When building an Ubuntu bdrv, use the default apt settings

dimkr committed:
fix syntax error in 431.svg

dimkr committed:
Merge pull request #2951 from dimkr/bugfix/ci-no-dc

Fix syntax error in 431.svg

dimkr committed:
if sudo is not installed, provide a dummy sudo

dimkr committed:
Merge pull request #2948 from dimkr/cleanup/xwin-without-xorg

If only Xwayland is present, start Xwayland and hide the X.Org button

dimkr committed:
Merge pull request #2942 from dimkr/feature/latest-initrd-progs

Switch to prebuilt initrd.gz binaries from GitHub

Commit:
Merge pull request #2939 from rizalmart/patch-411

hacks-postinstall2.sh: Some fixes

Commit:
Merge pull request #2937 from rizalmart/patch-409

removepreview.sh: Some improvements

Commit:
Merge pull request #2938 from rizalmart/patch-410

z_update_system_cache.sh: Some bugfixes

Commit:
Merge pull request #2936 from rizalmart/patch-408

installpkg.sh: Some improvements

dimkr committed:
Merge pull request #2950 from dimkr/feature/dummy-sudo

If sudo is not installed, provide a dummy sudo

dimkr committed:
prevent applications running as spot from reading files under /root

dimkr committed:
prevent applications running as spot from inheriting file descriptors

dimkr committed:
Merge pull request #2952 from dimkr/bugfix/spot-root-read

Prevent applications running as spot from reading files under /root

dimkr committed:
treat empty .xlockrc like a missing file and do not prompt for a password if -startCmd is specified

dimkr committed:
add GTK+ 4 theme support to ptheme

dimkr committed:
run lxtask on Ctrl+Shift+Esc

dimkr committed:
update l3afpad to 16f2222

dimkr committed:
Merge pull request #2959 from dimkr/feature/ctrl-shift-escape

Run lxtask on Ctrl+Shift+Esc

dimkr committed:
Merge pull request #2960 from dimkr/bugfix/l3afpad-master

Update l3afpad to 16f2222

dimkr committed:
Merge pull request #2958 from dimkr/feature/ptheme-gtk4

Add GTK+ 4 theme support to ptheme

dimkr committed:
Merge pull request #2953 from dimkr/bugfix/empty-xlockrc-no-suspend

Treat empty .xlockrc like a missing file and do not prompt for a password if -startCmd is specified

dimkr committed:
work around #2964 by using gtk2dialog

zigbert committed:
pMusic: Read Audio-CD

freedb.org was taken down in 2020, so we point instead to gnudb.org

zigbert committed:
pMusic: Play http radio stream without streamripper

zigbert committed:
pMusic: Move Streamripper from Recommended to Optional dependency

Streamripper is rather outdated, and many streams aren't working with it. The benefit of playing radio streams through Streamripper is to grab metainfo in real-time. In the future I guess we will be able to do this in ffmpeg itself by setting up a stream-server, but there is not much ducumentation around this topic...

zigbert committed:
pMusic: Bugfix: Podcast timestamp missed if https:

dimkr committed:
add a upup logo

dimkr committed:
fix syntax error in /etc/shells

dimkr committed:
Merge pull request #2966 from dimkr/feature/upup-logo

Add a upup logo

mavrothal committed:
Merge pull request #2968 from dimkr/bugfix/etc-shells-syntax

Fix syntax error in /etc/shells

dimkr committed:
allow the new hostname to be resolved over IPv4

dimkr committed:
Merge pull request #2965 from dimkr/bugfix/2964-workaround

Work around #2964 by using gtk2dialog

dimkr committed:
Merge pull request #2971 from dimkr/bugfix/puppypc-ipv6-only

Allow the new hostname to be resolved over IPv4

dimkr committed:
add support for periodic TRIM on the save partition

dimkr committed:
allow spot to read the cursor without access to /root/.icons

dimkr committed:
allow spot to read the GTK+ 2 theme without access to /root/.gtkrc-2.0

dimkr committed:
fix another instance of 0a2da1e

peabee committed:
PupClockset-2.8 & Pup-SysInfo-3.2

dimkr committed:
avoid the fstrim loop if /initrd/mnt/dev_save has no TRIM support

peabee committed:
git:// protocol has been removed

See https://forum.puppylinux.com/viewtopic.php?t=5440

dimkr committed:
Merge pull request #2976 from puppylinux-woof-CE/peabee-patch-1

git:// protocol has been removed

peabee committed:
Another git://

dimkr committed:
Merge pull request #2978 from puppylinux-woof-CE/peabee-patch-1

Another git://

mavrothal committed:
Merge pull request #2974 from dimkr/feature/periodic-trim

Add support for periodic TRIM on the save partition

peabee committed:
JWMDesk-3.5

dimkr committed:
always create ~/.icons when setting the cursor theme

dimkr committed:
drop the breeze-cursor-theme template

dimkr committed:
change the kbuild SFS file name to kbuild-`uname -r`.sfs

dimkr committed:
fix missing icons in pcur

dimkr committed:
before checking if a device supports TRIM, strip the partition number

dimkr committed:
Merge pull request #2975 from dimkr/bugfix/spot-themes

Fix appearance of applications running as spot since 916d661

dimkr committed:
Merge pull request #2980 from dimkr/bugfix/trim-partition-to-dev

Before checking if a device supports TRIM, strip the partition number

mavrothal committed:
Merge pull request #2979 from dimkr/feature/kbuild-uname-r-match

Change the kbuild SFS file name to kbuild-`uname -r`.sfs

dimkr committed:
auto-load bdrv and kbuild and hide them in sfs_load

mavrothal committed:
Merge pull request #2981 from dimkr/feature/bdrv-kbuild-auto

Auto-load bdrv and kbuild and hide them in sfs_load

rserwin1 committed:
Update hostname-set

rserwin1 commented:
The "Update hostname-set" commit should resolve the conflict.  Please tell me if there is anything else I need to do to allow the merge.

dimkr commented:
> The "Update hostname-set" commit should resolve the conflict. Please tell me if there is anything else I need to do to allow the merge.

The conflict is still there

rserwin1 commented:
I merged the conflicting line, "echo "::1 localhost $NEW_HOSTNAME" >> /tmp/hostname-set-hosts" into my changes.  The remaining "conflicts" are my updates to hostname-set, the indents and quotes around $HOSTNAME in two places.

I no longer see the message regarding a conflict, but forget how I actually found the original conflict message.

How do I notify github that the resolution is what I intend?

[edit] Now that I added a comment, I see the message.

[edit] I think I figured it out.  I edited the conflict listing and clicked on the "resolved" button.

dimkr commented:
You need to do a `git pull` (or `git fetch` + `git merge`), "merge" doesn't mean "take the same lines as in the other branch and putting them in my branch".

rserwin1 commented:
I am afraid that I do not understand how to perform the "git pull (or git fetch + git merge)".  Please assist me with that, now that you know my intent.

[edit] I need to be away for a while, so can resume this conversaion later.  BTW, thank you for your prompt replies.

dimkr commented:
You'll need to do some basic git tutorial and learn how to use git, which is not the same as using the GitHub web UI to edit files.

rserwin1 committed:
Merge branch 'network_connect_20220202' of https://github.com/rserwin1/woof-CE into network_connect_20220202

# Conflicts:
#	woof-code/rootfs-skeleton/usr/sbin/hostname-set

Resolved

rserwin1 commented:
I believe I have pushed the conflict resolution and do not now see any evidence of a current conflict.

Is there anything else I need to do for you to merge this PR?

dimkr commented:
Now this branch has 234 commits and 158 changed files.

I don't use SNS, so I have no objections other than git-related ones.

rserwin1 commented:
Only three of the commits are mine.  How did the others get added to my branch?

I did refresh my local repo testing branch several times while it held my branch.  Would the updates somehow get added to the branch instead of the testing branch?

What do I do now?  The only solution I can think of is to start over with a new branch and PR.  Is there any other way?

Also, I had to retrieve the remote copy of the branch to rebuild my local repo, which I had inadvertently deleted.  Maybe that pulled along the other commits.

rserwin1 commented:
Thanks for your patience with my effort, here.

Since I am unable to correct the "additional commits" problem with the network_connect_20220202 branch, I would like to start again from the beginning.  Would my deleting that branch (in github) and supplying a new branch delete my commits without impacting the added ones?  Maybe the PR needs to be rejected and closed, as well.  I am reluctant to guess on my own how to proceed, so please instruct me.

I am considering a replacement PR with a commit for the updated connectwizard and "cosmetized" versions of the other components.  Once they are merged I would submit a new PR -- or an addition to the replacement PR, if possible after the merge -- with the full updates of the remaining components.

That way, the critical fix to connectwizard gets done and the other components will reveal (in github) the substantive changes in a second commit, without the annoying indentation and command indicator (`...` -> $(...)) differences.

Please direct me so that I can get this done.

dimkr commented:
> I did refresh my local repo testing branch several times while it held my branch. Would the updates somehow get added to the branch instead of the testing branch?

It doesn't work like that, you need to `git pull`.

dimkr commented:
@rserwin1 I did that for you, see https://github.com/puppylinux-woof-CE/woof-CE/pull/3024.

If you want to get stared with git quickly and stop using GitHub's web UI,

```
git clone http://github.com/puppylinux-woof-CE/woof-CE -b network_connect_20220202
cd woof-CE
git remote add fork http://github.com/rserwin1/woof-CE
git branch network_connect_20220202-more-changes
git checkout network_connect_20220202-more-changes
# do your changes
git add file1 file2
git commit -m "commit message goes here"
git pus fork network_connect_20220202-more-changes
# open a pull request from rserwin1:network_connect_20220202-more-changes to puppylinux-woof-CE:network_connect_20220202
```

rserwin1 commented:
Thank you very much for cleaning up my original branch and giving me those instructions.

I have completed them except for:
open a pull request from rserwin1:network_connect_20220202-more-changes to puppylinux-woof-CE:network_connect_20220202

I assume that is to be a normal PR but with commentary that it supplements the original branch.  Before I do that, I want to be sure you don't mean more than that.

dimkr commented:
> I want to be sure you don't mean more than that.

Nope. Your changes are safe, they're in puppylinux-woof-CE/woof-CE:network_connect_20220202. Just open a PR with additional changes you want to add (don't forget to target that branch and not testing), and everything can be merged into testing together.

rserwin1 commented:
I may have over-responded by committing all but hostname-set.  It appeared to me that connectwizard had not been changed in the original branch, so concluded that only hostname-set is there, with the resolved conflict.  I think I have changed only connectwizard and rc.network_eth from the original branch.

Will it be OK to submit the 9 and assume the unchanged files are handled or ignored?

dimkr commented:
> Will it be OK to submit the 9 and assume the unchanged files are handled or ignored?

I'm not sure I understand. The changes in the "Files changed" view are the ones that make it into the destination branch, after a pull request is merged. If you don't want some of the changes to be merged, you'll need to revert them.

https://github.com/puppylinux-woof-CE/woof-CE/pull/3024/files

rserwin1 commented:
I double-checked my original branch in rserwin1 and see only hostname-set as having my changes and the conflict resolved.  The other do not have my changes.

So I will go  ahead and submit the commit of the 9 changed files.
